File: a2oj/ACB_round_1/soln.rb
t = gets.strip.to_i
for i in (0...t)
    contestnats = []
    qualified = 0
    n, top = gets.strip.split(' ')
    n = n.to_i
    top = top.to_i - 1
    for j in (0...n)
        contestnats << gets.strip.to_i
    end
    contestnats.sort!.reverse!
    while contestnats[top] == 0
        top -= 1
    end
    for m in (0...contestnats.length)
        if contestnats[m] >= contestnats[top]
            qualified += 1
        end
    end
    puts qualified
end

File: a2oj/ACB_round_1/soln.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main() {
    int t = 0, n = 0, top = 0, i = 0, j = 0, m = 0, qualified = 0;
    cin >> t;
    for (i = 0; i < t; i++) {
        cin >> n >> top;
        top = top - 1;
        qualified = 0;
        vector<int> contestnats(n);
        for(j = 0; j < n; j++) {
            cin >> contestnats[j];
        }
        sort(contestnats.begin(), contestnats.end());
        reverse(contestnats.begin(),contestnats.end());
        while(contestnats[top] == 0){
            top--;
        }
        for (m = 0; m < contestnats.size(); m++) {
            if(contestnats[m] >= contestnats[top]){
                qualified++;
            }
        }
        cout << qualified << endl;
    }
    return 0;
}
